cheroot/server.py

```diff
@@ -822,7 +823,9 @@ def read_request_line(self):
 
             # `urlsplit()` above parses "example.com:3128" as path part of URI.
             # this is a workaround, which makes it detect netloc correctly
-            uri_split = urllib.parse.urlsplit(b'//' + uri)
+            uri_split = urllib.parse.urlsplit(
+                '//{}'.format(uri)
+                .encode('ascii'))
```

Comment:
Please don't change logic here. Encode the part that needs to be encoded and keep the literal in place. I improved readability making guessing data types easier.